File: test_schedule_data.js
// Test script to check schedule data
// Chạy script này trong browser console khi đang ở trang schedule

console.log('=== CHECKING SCHEDULE DATA ===');

// 1. Check campuses
async function checkCampuses() {
    try {
        const response = await fetch('/api/campuses', {
            headers: {
                'Authorization': `Bearer ${localStorage.getItem('auth_token')}`
            }
        });
        const data = await response.json();
        console.log('CAMPUSES:', data);
        return data;
    } catch (error) {
        console.error('Error fetching campuses:', error);
        return [];
    }
}

// 2. Check classrooms
async function checkClassrooms() {
    try {
        const response = await fetch('/api/classrooms', {
            headers: {
                'Authorization': `Bearer ${localStorage.getItem('auth_token')}`
            }
        });
        const data = await response.json();
        console.log('ALL CLASSROOMS:', data);
        return data;
    } catch (error) {
        console.error('Error fetching classrooms:', error);
        return [];
    }
}

// 3. Check classrooms by campus
async function checkClassroomsByCampus(campusId) {
    try {
        const response = await fetch(`/api/classrooms?campus_id=${campusId}`, {
            headers: {
                'Authorization': `Bearer ${localStorage.getItem('auth_token')}`
            }
        });
        const data = await response.json();
        console.log(`CLASSROOMS FOR CAMPUS ${campusId}:`, data);
        return data;
    } catch (error) {
        console.error(`Error fetching classrooms for campus ${campusId}:`, error);
        return [];
    }
}

// 4. Check subjects
async function checkSubjects() {
    try {
        const response = await fetch('/api/subjects', {
            headers: {
                'Authorization': `Bearer ${localStorage.getItem('auth_token')}`
            }
        });
        const data = await response.json();
        console.log('SUBJECTS:', data);
        return data;
    } catch (error) {
        console.error('Error fetching subjects:', error);
        return [];
    }
}

// 5. Check teachers
async function checkTeachers() {
    try {
        const response = await fetch('/api/teachers', {
            headers: {
                'Authorization': `Bearer ${localStorage.getItem('auth_token')}`
            }
        });
        const data = await response.json();
        console.log('TEACHERS:', data);
        return data;
    } catch (error) {
        console.error('Error fetching teachers:', error);
        return [];
    }
}

// Main function
async function main() {
    console.log('Starting data check...');
    
    // Check all data
    const campuses = await checkCampuses();
    const classrooms = await checkClassrooms();
    const subjects = await checkSubjects();
    const teachers = await checkTeachers();
    
    // Check classrooms by campus
    for (const campus of campuses) {
        await checkClassroomsByCampus(campus.id);
    }
    
    // Summary
    console.log('=== SUMMARY ===');
    console.log(`Campuses: ${campuses.length}`);
    console.log(`Classrooms: ${classrooms.length}`);
    console.log(`Subjects: ${subjects.length}`);
    console.log(`Teachers: ${teachers.length}`);
    
    // Check classroom-campus relationships
    console.log('=== CLASSROOM-CAMPUS RELATIONSHIPS ===');
    const campusCounts = {};
    for (const classroom of classrooms) {
        const campusId = classroom.campus_id;
        if (campusId) {
            campusCounts[campusId] = (campusCounts[campusId] || 0) + 1;
        } else {
            campusCounts['NULL'] = (campusCounts['NULL'] || 0) + 1;
        }
    }
    
    for (const [campusId, count] of Object.entries(campusCounts)) {
        if (campusId === 'NULL') {
            console.log(`Not assigned to any campus: ${count}`);
        } else {
            const campus = campuses.find(c => c.id === campusId);
            const campusName = campus ? campus.name : 'Unknown';
            console.log(`${campusName} (${campusId}): ${count}`);
        }
    }
}

// Run the check
main();

File: test_schedule_data.py
#!/usr/bin/env python3
"""
Test script to check schedule data and display complete information
"""

import sys
import os
sys.path.append('backend')

from backend.supabase_client import get_supabase_client

def test_schedules():
    """Test and display complete schedule data"""
    try:
        supabase = get_supabase_client()
        
        # Get all schedules with related data
        result = supabase.table('schedules').select('''
            *,
            classrooms!inner(
                id,
                name,
                code,
                campus_id,
                campuses(
                    id,
                    name,
                    code
                )
            ),
            subjects!inner(
                id,
                name,
                code
            ),
            teachers!inner(
                id,
                users!inner(
                    full_name,
                    email
                )
            )
        ''').execute()
        
        print(f"=== SCHEDULE DATA ANALYSIS ===")
        print(f"Total schedules found: {len(result.data)}")
        print()
        
        if not result.data:
            print("No schedule data found in database.")
            return
        
        # Group by day of week
        days = ['Thứ 2', 'Thứ 3', 'Thứ 4', 'Thứ 5', 'Thứ 6', 'Thứ 7', 'Chủ nhật']
        
        for day_index in range(7):
            day_schedules = [s for s in result.data if s['day_of_week'] == day_index]
            if day_schedules:
                print(f"=== {days[day_index]} ===")
                for schedule in day_schedules:
                    print(f"ID: {schedule['id']}")
                    print(f"  Lớp học: {schedule['classrooms']['name'] if schedule['classrooms'] else 'N/A'}")
                    print(f"  Mã lớp: {schedule['classrooms']['code'] if schedule['classrooms'] else 'N/A'}")
                    print(f"  Môn học: {schedule['subjects']['name'] if schedule['subjects'] else 'N/A'}")
                    print(f"  Mã môn: {schedule['subjects']['code'] if schedule['subjects'] else 'N/A'}")
                    print(f"  Giáo viên: {schedule['teachers']['users']['full_name'] if schedule['teachers'] and schedule['teachers']['users'] else 'N/A'}")
                    print(f"  Email GV: {schedule['teachers']['users']['email'] if schedule['teachers'] and schedule['teachers']['users'] else 'N/A'}")
                    print(f"  Thời gian: {schedule['start_time']} - {schedule['end_time']}")
                    print(f"  Phòng học: {schedule['room'] or 'N/A'}")
                    print(f"  Cơ sở: {schedule['classrooms']['campuses']['name'] if schedule['classrooms'] and schedule['classrooms']['campuses'] else 'N/A'}")
                    print(f"  Mã cơ sở: {schedule['classrooms']['campuses']['code'] if schedule['classrooms'] and schedule['classrooms']['campuses'] else 'N/A'}")
                    print(f"  Ngày tạo: {schedule['created_at']}")
                    print("  ---")
                print()
        
        # Check for missing data
        print("=== DATA COMPLETENESS CHECK ===")
        missing_classroom = sum(1 for s in result.data if not s['classrooms'])
        missing_subject = sum(1 for s in result.data if not s['subjects'])
        missing_teacher = sum(1 for s in result.data if not s['teachers'] or not s['teachers']['users'])
        missing_campus = sum(1 for s in result.data if not s['classrooms'] or not s['classrooms']['campuses'])
        
        print(f"Schedules missing classroom data: {missing_classroom}")
        print(f"Schedules missing subject data: {missing_subject}")
        print(f"Schedules missing teacher data: {missing_teacher}")
        print(f"Schedules missing campus data: {missing_campus}")
        
    except Exception as e:
        print(f"Error: {e}")
        import traceback
        traceback.print_exc()

if __name__ == "__main__":
    test_schedules()
